### PR TITLE
Create Locks Challenge Mini-Game

### PR DESCRIPTION
**Pull request by apatel78**
Added code for locks challenge mini-game
Added support for shadcn
Fixed minor bugs with spreadpool
Removed unnecessary console.log statements

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ node_modules
 /build
 /public/build
 .env*
+test.txt
```
Comment (chrisparsons83) [inline .gitignore:6]: Do we need this anymore?

File: .project [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (chrisparsons83) [inline .project:1]: I don't care if we add this in, more just curious what this is 😂

File: app/components/ui/label.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react"
+import * as LabelPrimitive from "@radix-ui/react-label"
+import { cva, type VariantProps } from "class-variance-authority"
+
+import { cn } from "~/utils"
+
+const labelVariants = cva(
```
Comment (chrisparsons83) [inline app/components/ui/label.tsx:7]: Is this just putting those classes in the className in line 18? If so, we can just inline the classes there and get rid of class-variance-authority.

File: app/components/ui/radio-group.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import * as React from "react"
+import { CheckIcon } from "@radix-ui/react-icons"
+import * as RadioGroupPrimitive from "@radix-ui/react-radio-group"
+
+import { cn } from "~/utils"
+
+const RadioGroup = React.forwardRef<
```
Comment (chrisparsons83) [inline app/components/ui/radio-group.tsx:7]: It doesn't feel like we need forwardRef for these but also no reason to hold up the commit on this. I went ahead and made #44 to look at it later.

File: app/routes/games.locks-challenge.rules.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export default function GamesLocksChallengeStreamingRules() {
+    return (
+      <>
+        <h2>Locks Challenge Rules</h2>
+        <p>
+          Start the season with 0 points. Each week, you can select as many teams to win
```
Comment (chrisparsons83) [inline app/routes/games.locks-challenge.rules.tsx:6]: Going to try a rewrite of the rules for clarity: (how are ties broken?)

Each week, you get to pick the winner of as many games as you'd like to choose. At the end of the week, if none of the teams you selected lost, you get one point per team you selected that won.

However, if any of your teams lose in that week, you score zero points for that week.

Ties don't count as a loss, but you don't get any points for a win from them either.

The player that scores the most points over the course of the season wins. There are no penalties for not picking any teams in a week besides foregoing the opportunity to earn points that week.

**Review by chrisparsons83 — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (apatel78) [reply .gitignore:6]: no

File: .project (hunk 2 above)
Comment (apatel78) [reply .project:1]: lmao no idea why I made this change

File: app/components/ui/label.tsx (hunk 3 above)
Comment (apatel78) [reply app/components/ui/label.tsx:7]: no idea this is shadcn generated code

File: app/components/ui/radio-group.tsx (hunk 4 above)
Comment (apatel78) [reply app/components/ui/radio-group.tsx:7]: no idea this is shadcn generated code

File: app/routes/games.locks-challenge.rules.tsx (hunk 5 above)
Comment (apatel78) [reply app/routes/games.locks-challenge.rules.tsx:6]: lgtm, I was probably going to make them using chatgpt before the season starts. Could potentially break end of season ties by looking at which player had the most correct picks.

Comment (chrisparsons83) [reply app/routes/games.locks-challenge.rules.tsx:6]: That seems fine to me. Are we handling that in the standings sorting currently? If not, we can add that in separately.

Comment (apatel78) [reply app/routes/games.locks-challenge.rules.tsx:6]: not currently being handled but I will add that logic in before the season starts

**Pull request merged**